db/log_reader.cc

```diff
@@ -191,6 +191,36 @@ bool Reader::ReadRecord(Slice* record, std::string* scratch,
         }
         break;
 
+      case kBadRecordLen:
```

Comment:
The difference is the ReportCorruption, which the test cases verify.  Do you want me to merge it and use a conditional to decide which corruption string to reprot?